core/src/test/java/org/axonframework/common/lock/PessimisticLockFactoryTest.java

```diff
@@ -143,4 +145,66 @@ private Thread createThread(final CountDownLatch starter, final CountDownLatch c
             }
         });
     }
+
+    @Test(timeout = 5000, expected = LockAcquisitionFailedException.class)
+    public void testAquireBackoff() throws InterruptedException {
+        PessimisticLockFactory.BackoffParameters backoffConfig = new PessimisticLockFactory.BackoffParameters(
+                10,
+                -1,
+                0
+        );
+        final PessimisticLockFactory lockFactory = new PessimisticLockFactory(backoffConfig);
+        final CountDownLatch randevous = new CountDownLatch(1);
+        final CountDownLatch lockObtained = new CountDownLatch(1);
+        final AtomicReference<Exception> exceptionInThread = new AtomicReference<>();
+        final String id = "aggregateId";
+        new Thread(() -> {
+            try {
+                lockFactory.obtainLock(id);
+                lockObtained.countDown();
+                randevous.await();
+            } catch (Exception e) {
+                exceptionInThread.set(e);
+            }
+        }).start();
+        lockObtained.await();
+        try {
+            lockFactory.obtainLock(id);
+        } finally {
+            randevous.countDown();
+        }
+    }
+
+    @Ignore("Sleep is required to get the 2nd thread in the wait of the 2nd tryLock, ignoring since this is unreliable")
+    @Test(timeout = 5000, expected = LockAcquisitionFailedException.class)
+    public void testQueueBackoff() throws InterruptedException {
+        PessimisticLockFactory.BackoffParameters backoffConfig = new PessimisticLockFactory.BackoffParameters(
+                -1,
+                1,
+                10000
+        );
+        final PessimisticLockFactory lockFactory = new PessimisticLockFactory(backoffConfig);
+        final CountDownLatch randevous = new CountDownLatch(1);
+        final CountDownLatch threadStarted = new CountDownLatch(2);
+        final AtomicReference<Exception> exceptionInThread = new AtomicReference<>();
+        final String id = "aggregateId";
+        createThread(lockFactory, randevous, threadStarted, exceptionInThread, id);
+        createThread(lockFactory, randevous, threadStarted, exceptionInThread, id);
+        threadStarted.await();
+        Thread.sleep(100);
+        lockFactory.obtainLock(id);
+        randevous.countDown();
+    }
+
+    private void createThread(PessimisticLockFactory lockFactory, CountDownLatch randevous, CountDownLatch threadStarted, AtomicReference<Exception> exceptionInThread, String id) {
+        new Thread(() -> {
+            try {
+                threadStarted.countDown();
```

Comment:
I think you'd want to move this line down below the `obtainLock`. This will ensure these threads have claimed the lock before your test-thread can access it. You can then test the behavior of the lock from that thread.
Make sure you call `rendezvous.countDown()` (typo) in a `finally` block, so that the threads and locks are released, regardless of the test's outcome.

Reply:
The problem is that only one thread will claim the lock and reach that line. since I need an additional thread to 'fill the queue' the test will never complete.